src/components/GameBoard.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { Square, Row } from '../constants';
+import Game from '../tic-tac-toe';
+
+class GameBoard extends React.Component<any, any> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      board: [
+        [undefined, undefined, undefined],
+        [undefined, undefined, undefined],
+        [undefined, undefined, undefined],
+      ],
+      boardSize: 0,
+      gameStatus: null,
+    };
+  }
+
+  public handleChange = (event: React.FormEvent<HTMLInputElement>): void => {
+    this.setState({
+      boardSize: event.currentTarget.value,
+    });
+  };
+
+  public submitBoardSize = (event: any): void => {
+    event.preventDefault();
+    const newGame = new Game(Number(this.state.boardSize));
+    console.log('NEW GAME', newGame);
+    this.setState({
+      board: newGame.board,
+      gameStatus: 'In progress',
+    });
+  };
+
+  public render() {
+    const { board } = this.state;
+    return (
+      <div className='container center'>
+        <div className='center-container center column'>
+          <h1>Tic Tac Toe</h1>
+          <p>What size should the board be?</p>
+          <form onSubmit={this.submitBoardSize}>
+            Board Size:{' '}
+            <input onChange={this.handleChange} type='text' name='size' />
+            <input type='submit' value='submit' />
+          </form>
+          {this.state.gameStatus ? (
+            <table>
+              <tbody>
+                {board.map((row: Row, _idx: number) => (
+                  <tr key={_idx}>
```

Comment:
Your linter should hopefully warn on this: there's no point in (and bugs can come from) using the index as the component key.